pkg/api/pod/warnings_test.go

```diff
@@ -599,11 +600,120 @@ func TestWarnings(t *testing.T) {
 				`spec.affinity.podAntiAffinity.preferredDuringSchedulingIgnoredDuringExecution[1].podAffinityTerm.labelSelector: a null labelSelector results in matching no pod`,
 			},
 		},
+		{
+			name: "no duplicate container ports",
```

Comment:
for success cases you want to test:

* one container, no ports
* one container, one port
* one container, two ports, different ports, same protocol
* one container, two ports, same port, different protocol
* one container, two ports, same port, same protocol, different hostPort
* one container, two ports, same port, same protocol, same hostPort, different hostIP
* two containers, one porteach , different ports, same protocol
* two containers, one porteach , same port, different protocol
* two containers, one port each, same port, same protocol, different hostPort
* two containers, one port each, same port, same protocol, same hostPort, different hostIP

Reply:
> * one container, two ports, same port, same protocol, different hostPort
> * one container, two ports, same port, same protocol, same hostPort, different hostIP
>
> …
>
> * two containers, one port each, same port, same protocol, different hostPort
> * two containers, one port each, same port, same protocol, same hostPort, different hostIP

I thought containers of a pod share a network namespace. I didn't expect that one pod-port could relate to multiple host-ports. What happens to packets in these cases?

Reply:
```
one container, two ports, same port, same protocol, different hostPort
two containers, one port each, same port, same protocol, different hostPort
```
As hostIP is not set, these two cases will fail.

Reply:
nope, the hostPort is different.

```
$ k get pod iamyouare-744964d759-87g8b -o json | jq '.spec.containers[0].ports[]'
{
  "containerPort": 9376,
  "protocol": "TCP"
}
{
  "containerPort": 9376,
  "hostPort": 93,
  "protocol": "TCP"
}
{
  "containerPort": 9376,
  "hostPort": 76,
  "protocol": "TCP"
}
```

Reply:
> I thought containers of a pod share a network namespace. I didn't expect that one pod-port could relate to multiple host-ports. What happens to packets in these cases?

Yeah, I thought it was a bug until I tried to take it away and people yelled. It works, for the most part.

Reply:
https://github.com/kubernetes/kubernetes/pull/113245#discussion_r1260815743 This is the test case.

```
one container, two ports, same port, same protocol, different hostPort
two containers, one port each, same port, same protocol, different hostPort
```
With the curent change, those two cases will not fail.

Reply:
this should be OK